test/ipcFixtures.hpp

```diff
@@ -522,6 +522,44 @@ TEST_P(umfIpcTest, ConcurrentGetPutHandles) {
     EXPECT_EQ(stat.putCount, stat.getCount);
 }
 
+TEST_P(umfIpcTest, ConcurrentGetPutHandles) {
+    std::vector<void *> ptrs;
+    constexpr size_t ALLOC_SIZE = 100;
+    constexpr size_t NUM_POINTERS = 100;
+    umf::pool_unique_handle_t pool = makePool();
+    ASSERT_NE(pool.get(), nullptr);
+
+    for (size_t i = 0; i < NUM_POINTERS; ++i) {
+        void *ptr = umfPoolMalloc(pool.get(), ALLOC_SIZE);
+        EXPECT_NE(ptr, nullptr);
+        ptrs.push_back(ptr);
+    }
+
+    umf_test::syncthreads_barrier syncthreads(NTHREADS);
+
+    auto getPutHandlesFn = [&ptrs, &syncthreads](size_t) {
+        syncthreads();
+        for (void *ptr : ptrs) {
```

Comment:
It might be useful to have one test case where ptrs are shuffled differently for each thread

Reply:
Yeah, but my idea is to achieve high contention on the IPC cache when every thread opens the same IPC handle and tries to put it in the cache, but only one should succeed. If `ptrs` are shuffled then the probability of the contention is decreased.

Reply:
Done. So I implemented both version shuffled and not-shuffled